service/interpreter/workflowImpl.go

```diff
@@ -788,3 +790,32 @@ func createUserWorkflowError(provider WorkflowProvider, message string) error {
 		message,
 	)
 }
+
+func WaitForStateCompletionWorkflowImpl(ctx UnifiedContext, provider WorkflowProvider, input service.WaitForStateCompletionWorkflowInput) (*service.WaitForStateCompletionWorkflowOutput, error) {
+	signalName := input.StateCompletionSignalId
+	if signalName == "" {
+		return nil, createUserWorkflowError(provider, "state completion signal id is empty")
+	}
+
+	selector := provider.NewSelector(ctx)
+	signalValue := selector.ReceiveSignalValueBlocking(ctx, signalName)
+
+	if signalValue == nil {
+		return nil, createUserWorkflowError(provider, "state completion signal not found")
+	}
+
+	jsonData, err := json.Marshal(signalValue)
```

Comment:
Yeah like I mentioned above, the better way is to pass the pointer directly , like other APIs